packages/module-ui--selection-mode/src/lib/openlayers-selection-controller.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import { getWindow } from '@wme-enhanced-sdk/utils';
+import { SelectionFilter, SelectableFeatureType } from './types.js';
+import { WazeWindow, getLayerForFeatureType, getFeatureFromLayer, getFeaturesFromPermanentHazardLayers } from './waze-map-helpers.js';
+
+/**
+ * Controller for intercepting OpenLayers-level selection events
+ * 
+ * This provides comprehensive selection control by:
+ * - Intercepting direct user clicks on map features
+ * - Controlling hover states
+ * - Working across all scripts and selection methods
+ */
+export class OpenLayersSelectionController {
+  private _window: WazeWindow;
+  private _filter: SelectionFilter;
+  private _onAttemptSelect?: (featureType: SelectableFeatureType, featureId: number | string) => boolean;
+  private _originalSelectControl: any = null;
+  private _isActive = false;
+
+  constructor(
+    filter: SelectionFilter,
+    onAttemptSelect?: (featureType: SelectableFeatureType, featureId: number | string) => boolean
+  ) {
+    this._window = getWindow() as WazeWindow;
+    this._filter = filter;
+    this._onAttemptSelect = onAttemptSelect;
+  }
+
+  /**
+   * Activates the OpenLayers selection control
+   */
+  activate(): void {
+    if (this._isActive) return;
+    
+    const wazeMap = this._window.W?.map;
+    if (!wazeMap) {
+      console.warn('WME map not available for OpenLayers selection control');
+      return;
+    }
+
+    // TODO: Implement OpenLayers event interception
+    // This would involve:
+    // 1. Accessing W.selectionManager or equivalent
+    // 2. Intercepting feature selection events
+    // 3. Filtering based on our criteria
+    // 4. Preventing selection if feature doesn't match filter
```

Comment:
Kinda dissapointed you aborted on that. No worries, I'll change the base branch so all our changes won't get into the main branch, and I'll work on a seperate one to make it work.